### PR TITLE
[SCB-728] Decrypt/signature and adjust form in edge

### PR DESCRIPTION
**Pull request by wujimin**
1.add "decrypt“ logic, decrypt field named "body" in form,  generate new multiple fields to form by "body"
2.move "signature" to edge, and only signature for one operation
3.request with form-data or x-www-form-urlencoded, edge convert to json automatically
4.query userId by serviceToken, and put userId back to form
5.calc signature from response buffer,  and add "signature" field to response body, eg:
  before: {"k":"v"}
  after:    {"k":"v","signature":"xxxxx"}

### CODE REVIEW

File: demo/demo-edge/consumer/src/main/java/org/apache/servicecomb/demo/edge/consumer/Consumer.java [hunk 1]
```diff
@@ -111,6 +114,33 @@ public void run(String prefix) {
...
+
+    HttpEntity<MultiValueMap<String, String>> entity = new HttpEntity<>(form, headers);
+
+    @SuppressWarnings("unchecked")
+    Map<String, Object> result = (Map<String, Object>) template.postForObject(url, entity, Map.class);
+    Assert.isTrue(result.containsKey("signature"), "must exist signature");
+    result.remove("signature");
+
+    String expected = "{name=userName, age=10, userId=serviceTokenTest-userId, body1=b1, body2=b2, body3=b3}";
+    System.out.println("encrypt result: " + result.toString());
```
Comment (WillemJiang) [inline demo/demo-edge/consumer/src/main/java/org/apache/servicecomb/demo/edge/consumer/Consumer.java]: We are not supposed to use System.out.println in the test.

File: demo/demo-edge/edge-service/src/main/java/org/apache/servicecomb/demo/edge/service/encrypt/filter/EdgeSignatureResponseFilter.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  @Override
+  public void beforeSendResponse(Invocation invocation, HttpServletResponseEx responseEx) {
+    EncryptContext encryptContext = (EncryptContext) invocation.getHandlerContext().get(EdgeConst.ENCRYPT_CONTEXT);
+    if (encryptContext == null) {
+      return;
+    }
+    Hcr hcr = encryptContext.getHcr();
+
+    // bad practice: it's better to set signature in response header
```
Comment (WillemJiang) [inline demo/demo-edge/edge-service/src/main/java/org/apache/servicecomb/demo/edge/service/encrypt/filter/EdgeSignatureResponseFilter.java]: Is it a stand request or just some specific user requirement.

File: demo/demo-edge/edge-service/src/main/resources/META-INF/services/org.apache.servicecomb.common.rest.filter.HttpServerFilter [hunk 3]
```diff
@@ -0,0 +1,4 @@
+org.apache.servicecomb.demo.edge.service.encrypt.filter.EdgeSignatureRequestFilter
```
Comment (WillemJiang) [inline demo/demo-edge/edge-service/src/main/resources/META-INF/services/org.apache.servicecomb.common.rest.filter.HttpServerFilter]: License header please.

File: demo/demo-edge/edge-service/src/main/java/org/apache/servicecomb/demo/edge/service/encrypt/filter/EdgeSignatureResponseFilter.java (hunk 2 above)
Comment (wujimin) [reply demo/demo-edge/edge-service/src/main/java/org/apache/servicecomb/demo/edge/service/encrypt/filter/EdgeSignatureResponseFilter.java]: it's a demo for one specific user
it's a demo, not product.

File: demo/demo-edge/consumer/src/main/java/org/apache/servicecomb/demo/edge/consumer/Consumer.java (hunk 1 above)
Comment (wujimin) [reply demo/demo-edge/consumer/src/main/java/org/apache/servicecomb/demo/edge/consumer/Consumer.java]: done

File: demo/demo-edge/edge-service/src/main/resources/META-INF/services/org.apache.servicecomb.common.rest.filter.HttpServerFilter (hunk 3 above)
Comment (wujimin) [reply demo/demo-edge/edge-service/src/main/resources/META-INF/services/org.apache.servicecomb.common.rest.filter.HttpServerFilter]: done

**Pull request merged**